### DISCUSSION TITLE
Creating a LoRa of Your Own Data

### DISCUSSION THREAD

**Discussion by olihough86** [oobabooga/text-generation-webui#469] [General]
I'm knocked back at the speed things have developed int he past couple of weeks. 

Now with the ability to create Alpaca as a LoRA which is great, how much of a step is this from being able to create LoRAs with a bunch of your own data? for example feeding in a whole bunch of papers you wrote or niche subjects that you know the base models will be lacking on, or even trying to push the model to lean towards a certain writing style or subject? I'm seeing a lot of guides on instruct fine tuning but nothing on just cramming in more text.

So I guess my question is, can someone put together a short guide for those of us that don't have any experience training models? (I'm very tech savvy, but many AI subjects are witchcraft to me)

**Comment by olihough86** [comment]
Looks like I'm not the only one :D seems progress was made here 

https://github.com/tloen/alpaca-lora/issues/45

**Comment by oobabooga** [comment]
See here, it's super easy to create a LoRA: https://github.com/oobabooga/text-generation-webui/wiki/Using-LoRAs#training-a-lora

Formatting and cleaning the dataset is the hardest part.

**Comment by olihough86** [comment]
Thanks, I might have got the wrong end of the stick, this assumes you have your data as questions and answers, I guess I could generate a bunch of question answer pairs based off my data set, do you think this would be the best method of getting new knowledge in?

**Comment by vdruts** [comment]
Is question and answer data-pairs the only way to fine-tune? Is there anyway of having the LLM parse and make sense of data?

**Comment by oobabooga** [comment]
The line that generates the training data is this one: https://github.com/tloen/alpaca-lora/blob/main/finetune.py#L164

You could try using a single row with a very long string (like a book) to see if it works. I am not experienced with training neural networks, hopefully someone can tell us the right way to do this.

**Comment by St33lMouse** [comment]
It would be really cool to train a lora based on a novel. For example, you train on Lord of the Rings, then load attach the lora to your favorite adventure model and boom! 

You've got a ready made adventure with strong knowledge of Lord of the Rings.  Could this work? Is it working already, or has anyone tried it?
